sickbeard/dailysearcher.py

```diff
@@ -79,19 +79,15 @@ def run(self, force=False):
                 logger.log(u"ERROR: expected to find a single show matching " + str(sqlEp['showid']))
                 continue
 
-            try:
-                end_time = sbdatetime.sbdatetime.convert_to_setting(network_timezones.parse_date_time(sqlEp['airdate'], show.airs,
-                                                             show.network)) + datetime.timedelta(
-                    minutes=helpers.tryInt(show.runtime, 60))
-                #Keep this for future debug
-                #logger.log(u"Show %s ends at %s and now it is %s. Runtime is %s and airs %s" % (show.name, end_time, curTime, show.runtime, show.airs),logger.DEBUG )
-
-                # filter out any episodes that haven't aried yet
-                if end_time > curTime:
+            if show.airs and show.network:
+                # This is how you assure it is always converted to local time
+                air_time = network_timezones.parse_date_time(sqlEp['airdate'], show.airs, show.network).astimezone(network_timezones.sb_timezone)
+
+                # filter out any episodes that haven't started airing yet,
+                # but set them to the default status while they are airing
+                # so they are snatched faster
+                if air_time > curTime:
```

Comment:
@miigotu why did you changed the logic here?

"set them to the default status while they are airing": so as soon as DAILYSEARCHER set to WANTED, SEARCHQUEUE will immediately starts to search for them while show is still airing. Makes no sense to make to searchs while show hasn't finished yet.